state/transaction.go

```diff
@@ -607,6 +607,17 @@ func (s *State) internalProcessUnsignedTransaction(ctx context.Context, tx *type
 		previousBatch = lastBatches[1]
 	}
 
+	timestamp := uint64(lastBatch.Timestamp.Unix())
+
+	latestL2BlockNumber, err := s.PostgresStorage.GetLastL2BlockNumber(ctx, dbTx)
```

Comment:
This must be called only if l2BlockNumer != nil, the we can put this function call and the "if" of the line 617 inside an "if l2BlockNumber != nil" block (and remove the "l2BlockNumer != nil" from line 617)